Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py

```diff
@@ -652,12 +659,17 @@ def ResetNegatives2D(self, wsName, addMin, resetValue):
                             DataY=yDataNew,
                             DataE=eData,
                             NSpec=mtd[wsName].getNumberHistograms(),
+                            UnitX='dSpacing',
                             ParentWorkspace=mtd[wsName])
+            mtd[wsName].setDistribution(True)
```

Comment:
It's a small thing but `ADS.retrieve(wsName)` is preferred to mtd . You can import it like so
`from mantid.api import AnalysisDataService as ADS`